docs/adrs/024-remote-module-invocation.md

```diff
@@ -0,0 +1,280 @@
+information about a specific method and includes things like the CaikitMethodSignature, 
+request&response DataModel names, and the remote RPC name. The `RemoteMethodRpc` dataclass contains 
+all the runtime-specific assumptions  and is the main point where overlap happens. The 
+`RemoteModelFinder`, which constructs `RemoteMethodRpc`s, should not import `caikit.runtime`; 
+however, it will re-use a lot of code around the different naming schemes.
+
+We created the following pseudocode to help illustrate how the `RemoteModelFinder` constructs a 
+`RemoteModuleRpc`. (Note this is the function used in the pseudocode for the [RemoteModelFinder](#remotemodelfinder))
+
+```python
+def generate_config_for_module(module: ModuleBase, connection_info: Dict[str, Any], model_path: str)->RemoteModuleConfig:
+    # Gather a description of all tasks and their associated methods
+    task_methods = []
+    for task_class in module.tasks:
+        task_functions = []
+        for input, output, signature in module_class.get_inference_signatures(task_class):
+            # Construct request_dm_name,  task_request_name, and rpc_name. This makes assumptions
+            # about caikit.runtime service generation
+            request_dm_name ~= "TaskRequest"
+            task_request_name ~= "TaskPredict"
+            rpc_name = "/api/v1/TaskPredict"
+            
+            task_functions.append(
+                RemoteMethodRpc(
+                    signature=signature,
+                    request_dm_name=request_dm_name,
+                    response_dm_name=signature.return_type.__name__,
+                    rpc_name=rpc_name,
+                    input_streaming=input,
+                    output_streaming=output,
+                )
+            )
+            
+        task_functions.append((task_class, task_functions))
+    
+    # Gather description of the Train functions
+    train_method = None
+    if module.TRAIN_SIGNATURE:
+        request_dm_name ~= "TrainRequest"
+        rpc_name ~= "/api/v1/TaskTrain"
+        
+        train_method = RemoteMethodRpc(
+            signature=module.TRAIN_SIGNATURE,
+            request_dm_name=request_dm_name,
+            response_dm_name=module.TRAIN_SIGNATURE.return_type.__name__,
+            rpc_name=rpc_name,
+        )
+    
+    # Construct the remote config
+    return RemoteModuleConfig(
+        {
+            # Connection info
+            "connection":connection_info,
+            # Method info
+            "task_methods":task_methods,
+            "train_method":train_method,
+            # Source Module Information
+            "model_path": model_path,
+            "module_id": module.MODULE_ID,
+            "module_name": module.MODULE_NAME,
+        }
+    )
+        
+```
+
+### Diagram
+
+<img width="1439" alt="image" src="https://github.com/HonakerM/MTConnect-Python-Agent/assets/37811263/70563a23-8439-4221-b81d-c242f638a579">
```

Comment:
This is a really nice diagram! I think the one thing that feels a little confusing about it is how the `remote` versions seem like they're their own set of sequential things. I know that's not the intent, but it would be good to clarify that the `finder`/`initializer` boundaries are still encapsulated, so it would be critical for a given model to only be found by a single `finder` and in the case of the `RemoteModelFinder`, that would need to happen IFF the `RemoteModelInitializer` would be guaranteed to actually be able to initialize it.